### PR TITLE
EZP-30281: Added deprecation info to Core RichText FT namespace

### PR DESCRIPTION
**Pull request by MagdalenaZuba**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://jira.ez.no/browse/EZP-30281
| Versions      | 2.4+

### CODE REVIEW

File: docs/api/field_type_reference.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ Custom Field Types have to be programmed in PHP. However, the built-in Field Ty
 | [Rating](#rating-field-type) | **Deprecated** | N/A | N/A |
 | [Relation](#relation-field-type) | Validates and stores a relation to a Content item. | Yes, with both Field and FieldRelation criterions | Yes |
 | [RelationList](#relationlist-field-type) | Validates and stores a list of relations to Content items. | Yes, with FieldRelation criterion | Yes |
-| [RichText](#richtext-field-type) | Validates and stores structured rich text in DocBook xml format, and exposes it in several formats. | Yes[^1^](#1-note-on-legacy-search-engine)  | Yes |
+| [RichText](#richtext-field-type) | **Deprecated** |N/A| N/A |
```
Comment (alongosz) [inline docs/api/field_type_reference.md]: This is not correct, namespace was deprecated and now separate package provides FT, but the FT itself is not deprecated

File: docs/api/field_type_reference.md [hunk 2]
```diff
@@ -1881,15 +1881,19 @@ $validators = [
 
 ## RichText Field Type
 
+!!! caution "Deprecated"
+
+    The RichText Field Type is deprecated since v2.4. Relying on any class from the `eZ\Publish\Core\FieldType` namespace is deprecated.
```
Comment (alongosz) [inline docs/api/field_type_reference.md]: This is half-correct, I think entire section needs to be rephrased, starting from the information that RichText Field Type is provided via eZ Platform RichTextBundle and so it should be enabled.
Then we should mention that prior version 2.4 it was provided by Kernel via `eZ\Publish\Core\FieldType\RichText` namespace which is now deprecated.

File: docs/api/field_type_reference.md [hunk 3]
```diff
@@ -1881,15 +1881,19 @@ $validators = [
...
+
+    If you're implementing any interface or extending any base class from that namespace, refer to its PHPDoc to see what to implement or extend instead.
```
Comment (alongosz) [inline docs/api/field_type_reference.md]: This sentence is ok, should be connected to the message above.

File: docs/api/field_type_reference.md [hunk 4]
```diff
@@ -1881,15 +1881,19 @@ $validators = [
...
+    Make sure to enable the eZ Platform RichTextBundle provided by [ezsystems/ezplatform-richtext package](https://github.com/ezsystems/ezplatform-richtext).
```
Comment (alongosz) [inline docs/api/field_type_reference.md]: This is ok, to be merged with the starting information mentioned 2 comments above.

File: docs/releases/ez_platform_v2.4.md [hunk 5]
```diff
@@ -28,7 +28,11 @@
 
 #### RichText Field Type
 
-RichText Field Type has been extracted to a separate bundle, [ezplatform-richtext](https://github.com/ezsystems/ezplatform-richtext).
+RichText Field Type has been extracted to a separate bundle, [ezsystems/ezplatform-richtext](https://github.com/ezsystems/ezplatform-richtext). Relying on any class from the `eZ\Publish\Core\FieldType` namespace is deprecated.
```
Comment (alongosz) [inline docs/releases/ez_platform_v2.4.md]:
```suggestion
RichText Field Type has been extracted to a separate bundle, [ezsystems/ezplatform-richtext](https://github.com/ezsystems/ezplatform-richtext). Relying on any class from the `eZ\Publish\Core\FieldType\RichText` namespace is deprecated.
```

File: docs/api/field_type_reference.md (hunk 3 above)
Comment (MagdalenaZuba) [reply docs/api/field_type_reference.md]: It is, it's in the same caution message, but it is more readable/has more impact with a small space above.

File: docs/api/field_type_reference.md [hunk 6]
```diff
@@ -36,7 +36,7 @@ Custom Field Types have to be programmed in PHP. However, the built-in Field Ty
 | [Rating](#rating-field-type) | **Deprecated** | N/A | N/A |
 | [Relation](#relation-field-type) | Validates and stores a relation to a Content item. | Yes, with both Field and FieldRelation criterions | Yes |
 | [RelationList](#relationlist-field-type) | Validates and stores a list of relations to Content items. | Yes, with FieldRelation criterion | Yes |
-| [RichText](#richtext-field-type) | Validates and stores structured rich text in DocBook xml format, and exposes it in several formats. | Yes[^1^](#1-note-on-legacy-search-engine)  | Yes |
+| [RichText](#richtext-field-type) | Validates and stores structured rich text in DocBook xml format, and exposes it in several formats. Available via eZ Platform RichTextBundle. | Yes[^1^](#1-note-on-legacy-search-engine)  | Yes |
```
Comment (DominikaK) [inline docs/api/field_type_reference.md]:
```suggestion
| [RichText](#richtext-field-type) | Validates and stores structured rich text in DocBook xml format, and exposes it in several formats. Available via [eZ Platform RichTextBundle](https://github.com/ezsystems/ezplatform-richtext). | Yes[^1^](#1-note-on-legacy-search-engine)  | Yes |
```

File: docs/api/field_type_reference.md [hunk 7]
```diff
@@ -1881,15 +1881,18 @@ $validators = [
 
 ## RichText Field Type
 
+!!! caution "Deprecated"
+
+    Make sure to enable new version of the RichText Field Type provided via [eZ Platform RichTextBundle](https://github.com/ezsystems/ezplatform-richtext).
+    
+    The RichText Field Type provided by Kernel via `eZ\Publish\Core\FieldType\RichText` namespace is deprecated. Refer to PHPDoc whenewer you want to implement any interface or extend any base class from that namespace.
```
Comment (DominikaK) [inline docs/api/field_type_reference.md]:
```suggestion
    The RichText Field Type provided by Kernel via the `eZ\Publish\Core\FieldType\RichText` namespace is deprecated. Refer to PHPDoc whenever you want to implement any interface or extend any base class from that namespace.
```

**Pull request merged**